src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingtoolsapi/service/DefinitionService.kt

```diff
@@ -17,20 +18,19 @@ class DefinitionService(
 
   fun validateAndSave(
     definition: ProductDefinition,
-    caseLoads: List<String>,
+    authenticationToken: DprAuthAwareAuthenticationToken,
   ) {
     try {
       // Attempt mapping to assert references are correct
-      mapper.map(definition, RenderMethod.HTML, 1, caseLoads)
+      mapper.map(definition, RenderMethod.HTML, 1, authenticationToken)
 
       // Check each query executes successfully
       definition.dataset.forEach { dataset ->
         dataRepository.count(
           reportId = definition.id,
           filters = emptyList(),
-          caseloadFields = dataset.schema.field.filter { it.caseload }.map { it.name },
           query = dataset.query,
-          userCaseloads = caseLoads,
+          policyEngineResult = "FALSE",
```

Comment:
This will return a count of zero, but I guess it doesn't matter since the purpose of calling this is to make sure there is no error thrown, correct?

Reply:
Exactly!